Source/WebCore/rendering/svg/SVGTextLayoutEngineSpacing.cpp

```diff
@@ -36,14 +36,15 @@ SVGTextLayoutEngineSpacing::SVGTextLayoutEngineSpacing(const FontCascade& font)
 
 float SVGTextLayoutEngineSpacing::calculateCSSSpacing(const UChar* currentCharacter)
 {
-    const UChar* lastCharacter = m_lastCharacter;
+    static constexpr UChar nullFallback = 0;
+    const UChar* lastCharacter = m_lastCharacter ? m_lastCharacter : &nullFallback;
     m_lastCharacter = currentCharacter;
 
     if (!m_font.letterSpacing() && !m_font.wordSpacing())
         return 0;
 
     float spacing = m_font.letterSpacing();
-    if (currentCharacter && lastCharacter && m_font.wordSpacing()) {
```

Comment:
Is currentCharacter always non-zero? I assumed that, because you removed the check for it. But then we should change the function parameter to be a reference.